### ISSUE TITLE
Limits of hiragana-based romanisation

### ISSUE THREAD

**Issue by pipping** [tshatrov/ichiran#4]
Hi,

(this doesn't really belong in a bug report but I'd still like to take a second that I say what you've done here is fabulous, amazing, and incredibly helpful. Thank you!).

I'm not sure I understand completely what goes on in romanize.lisp, but under certain circumstances, it ends up merging an "o" and a "u" that it shouldn't. This issue is mentioned here: https://en.wikipedia.org/wiki/Nihon-shiki_romanization and 追う is given as an example. The correct reading of 追 is お, so that in hiragana, the word comes out as おう. This transformation is lossy/ambiguous, however: Here, お and う are pronounced separately, in contrast to 王, which, too, is romanised as おう but pronounced as a long お. To romanise 追う as ō is misleading, I think.

I believe that the general rule (and this might make for an easy fix) is: Merging of お and う cannot occur across kanji boundaries. In the presence of kanji, the breakup into hiragana and merging of お and う needs to occur before those tokens are thrown together.

Since I'm not a native speaker (quite the opposite), I checked forvo.com and found a recording that supports the claim that お and う are not joined in 追う: In the recording at http://forvo.com/word/%E8%BF%BD%E3%81%86/#ja by the user strawberrybrown, the お and the う can be made out quite distinctly. In contrast, I found a few examples of もう, ぽう, ちょう, and どう that she pronounces as mō, pō, chō, and dō, respectively, just as expected:

http://forvo.com/word/%E3%82%82%E3%81%86%E4%B8%80%E3%81%A4/
http://forvo.com/word/%E5%BD%93%E3%81%A6%E3%81%9A%E3%81%A3%E3%81%BD%E3%81%86/
http://forvo.com/word/%E3%81%A1%E3%82%87%E3%81%86%E3%81%A9%E3%82%88%E3%81%8B%E3%81%A3%E3%81%9F/
http://forvo.com/word/%E3%81%A9%E3%81%86%E3%81%8B%E3%81%97%E3%81%A6%E3%82%8B%E3%81%9C/

Which is to say, this user does not generally pronounce お and う sounds separately (as could be the case in a dialect, maybe?) but only when they're really meant to be separate.

Unfortunately and confusingly, there is another recording by the user smime also at http://forvo.com/word/%E8%BF%BD%E3%81%86/#ja which gives a pronunciation of 追う that is from my understanding incorrect. I would like to get to the bottom of this matter but do not know how.

Finally, please see also

https://en.wiktionary.org/wiki/%E8%BF%BD%E3%81%86#Verb (追う romaji: ou)
https://en.wiktionary.org/wiki/%E7%8E%8B#Noun_2 (王 romaji ō)

**Comment by tshatrov** [comment]
Yeah, this sounds like a good idea. One problem though, in JMdict database hiragana readings are not separated by kanji so this wasn't possible to implement at the time I wrote romanization algorithm. More recently I have implemented a kanji module (kanji.lisp) that has a function ```match-readings``` that can potentially be used to resolve this issue.

**Comment by pipping** [comment]
Oh. I didn't know that. I guess that makes what I had in mind quite difficult.

Special readings could be a problem. And then, what if (this is entirely fictional. I don't know if a real-world example exists) you have a word made up of two kanji, the first one could be read あ or あお and the latter can be read お or おう? If you only know that the entire word reads あおう, then that could be split into あ-おう or あお-う...

**Comment by tslater** [comment]
I noticed that the traditional hepburn option on the site doesn't create the `ō`. I'm wondering @tshatrov , is there a way to change the romanization settings using ichiran-cli?

**Comment by tshatrov** [comment]
@tslater I think if you do `(setf ichiran:*default-romanization-method* ichiran:*hepburn-basic*)` _before_ building the executable, then `-f` will use basic romanization.

**Comment by tslater** [comment]
Cool, I'll try it. Thanks!

**Comment by tslater** [comment]
Looks like it is working. Thanks!
